src/main/scala/symsim/examples/concrete/cartpole/CartPole.scala

```diff
@@ -0,0 +1,171 @@
+package symsim
+package examples.concrete.cartpole
+
+import symsim.concrete.Randomized
+
+/** A balancing cartpole example agent (with discretization)
+ *
+ * Based on: Razvan V. Florian.
+ * Correct equations for the dynamics of the cart-pole system.
+ * https://coneural.org/florian/papers/05_cart_pole.pdf
+ */
+
+case class CartPoleState (cp: Double, cv: Double, pa: Double, pv: Double):
+  override def toString: String = s"[cart position=$cp, cart velocity=$cv," +
+          s"pole angle=$pa, pole angular velocity=$pv]"
+
+type CartPoleObservableState = CartPoleState
+type CartPoleAction = Int
+type CartPoleReward = Double
+
+def closest (value: Double) (cutPoints: List[Double]): Double =
+  cutPoints
+    .find { value <= _ }
+    .getOrElse (cutPoints.last)
+
+val CP_MIN = -4.8
+val CP_MAX = 4.8
+val CV_MIN = -Double.MaxValue
+val CV_MAX = Double.MaxValue
+val PA_MIN = -0.418
+val PA_MAX = 0.418
+val PV_MIN = -Double.MaxValue
+val PV_MAX = Double.MaxValue
+
+val cPCutPoints = List (CP_MIN, -2.4, -1.0, -0.5,-0.2, -0.1, 0, 0.1, 0.2, 0.5, 1.0, 2.4, CP_MAX)
+val cVCutPoints = List (CV_MIN, -2, -1.5, -1.0, -0.7, -0.5, 0, 0.5, 0.7, 1.0, 1.5, 2, CV_MAX)
+val pACutPoints = List (PA_MIN, -0.2, -0.15, -0.1, -0.05, 0, 0.05, 0.1, 0.15, 0.2, PA_MAX)
+val pVCutPoints = List (PV_MIN, -2, -1.5, -1.0, -0.7, -0.5, 0, 0.5, 0.7, 1.0, 1.5, 2, PV_MAX)
+
+object CartPole
+  extends Agent[CartPoleState, CartPoleObservableState, CartPoleAction, CartPoleReward, Randomized]
+  with Episodic:
+
+    /** The episode should be guaranteed to terminate after
+      * TimeHorizon steps. This is used *only* *for* testing. It does
+      * not actually termintae the episodes. It is a bug if they run
+      * longer.
+      */
+    val TimeHorizon: Int = 2000
+
+    val gravity = 9.8
+    val masscart = 1.0
+    val masspole = 0.1
+    val total_mass = masspole + masscart
+    val length = 0.5
+    val polemass_length = masspole * length
+    val force_mag = 10.0
+    val tau = 0.02
```

Comment:
@mohsen-ghaffari1992 what is tau? Is it the time granularity ? What is the unit? seconds?

In general, are the units known for all parameters? or are they abstract quantities?

Reply:
@wasowski,
tau is the seconds between state updates.
I used openAI implementation variables: https://github.com/openai/gym/blob/master/gym/envs/classic_control/cartpole.py

Unfortunately, I did not understand the second part of you question correctly.

Reply:
Thanks, this will do for now.